### PR TITLE
4032 - Users: Get Many through Country User Summary

### PR DESCRIPTION
**Pull request by sorja**
- [x] All features/filters work
- [x] Use params vs string concatenation
- [x] Investigate slow performance (maybe missing index)
- [x] Include users get count: https://github.com/openforis/fra-platform/pull/4103/commits/a5773c728271dcebf591ba3cf68988081b08f625

Upcoming PRs:

1 update admin UI
2. update country home UI

### CODE REVIEW

File: src/meta/user/countryUserSummary.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { User } from 'meta/user/user'
+import { UserInvitation } from 'meta/user/userInvitation'
+
+export type CountryUserSummary = User & {
+  readonly fullName: string
+  readonly invitations: Array<UserInvitation>
```
Comment (minotogna) [inline src/meta/user/countryUserSummary.ts:6]: This object should have country iso and one optional invitation ?

File: src/server/repository/public/user/getMany.ts [hunk 2]
```diff
@@ -67,76 +54,46 @@ export const buildGetManyQuery = (props: UsersGetManyProps): BuildQueryReturned
...
-      and
-      `)}
-      group by ${selectFields}
-                   ${order}
-                   ${limit ? `limit ${limit}` : ''}
-                   ${offset ? `offset ${offset}` : ''}
+  with filtered_users as (
+    select distinct id
+    from ${schemaName}.country_user_summary cus
+    where 1 = 1
```
Comment (minotogna) [inline src/server/repository/public/user/getMany.ts]: Why this ? The where clauses could be build in an array and joined with and .

Also , does this query works with current admin filters ?

Comment (sorja) [reply src/server/repository/public/user/getMany.ts]: This query works with current admin filters

It was added for readability and separating all where clauses, 1=1 always return true and we can concatenate with ` and <clause>`

I will update

File: src/meta/user/countryUserSummary.ts (hunk 1 above)
Comment (sorja) [reply src/meta/user/countryUserSummary.ts:6]: This is something I am also investigating now that you mentioned it

In country collaborator view - yes, this is what we want: a single user, for single country with a single role

In admin though, we have one row with multiple country+role, so it makes sense to extend the user (although, we don't really extend all fields, but this is so we can use eg. Users.getRole)

File: src/server/repository/public/user/getMany.ts (hunk 2 above)
Comment (minotogna) [reply src/server/repository/public/user/getMany.ts]: Ok I see 😉it looked strange . I'll check the full file , not the diffs here .

**Comment by sorja** [comment]
@minotogna 

I have updated this PR, but I leave it in draft until tomorrow ( i want to double check it twice ).

The request takes now 108ms in local, down from 4s500ms (about 41x faster than initial)

The filtering including multiple filters is tested and working

**Review by minotogna — APPROVED** [review]

**Pull request merged**